tiproxy: add doc for load balance (#17874)
pingcap/docs: tiproxy/tiproxy-load-balance.md

tidb_servers:
  - host: tidb-host-1
    config:
      labels:
        zone: east
  - host: tidb-host-2
    config:
      labels:
        zone: west
tikv_servers:
  - host: tikv-host-1
    port: 20160
  - host: tikv-host-2
    port: 20160
  - host: tikv-host-3
    port: 20160
```

In the preceding configuration, the TiProxy instance on `tiproxy-host-1` prioritizes routing requests to the TiDB server on `tidb-host-1`, and the TiProxy instance on `tiproxy-host-2` prioritizes routing requests to the TiDB server on `tidb-host-2`.

## Connection count-based load balancing

TiProxy migrates connections from a TiDB server with more connections to those with fewer connections. This policy is not configurable and has the lowest priority.

Typically, TiProxy identifies the load on TiDB servers based on CPU usage. This policy usually takes effect in the following scenarios:

- When the TiDB cluster is just starting up, the CPU usage of all TiDB servers is close to 0. In this case, this policy prevents imbalanced load during startup.
- When the [CPU-based load balancing](#cpu-based-load-balancing) is not enabled, this policy ensures load balancing.

## Configure load balancing policies

> TiProxy supports configuring the combination and priority of the preceding load balancing policies through the configuration item [`policy`](/tiproxy/tiproxy-configuration.md#policy).

Reviewer: ```suggestion
TiProxy lets you configure the combination and priority of load balancing policies through the [`policy`](/tiproxy/tiproxy-configuration.md#policy) configuration item.
```

- `resource`: the resource priority policy performs load balancing based on the following priority order: status, health, memory, CPU, location, and connection count.
- `location`: the location priority policy performs load balancing based on the following priority order: status, location, health, memory, CPU, and connection count.

> - `connection`: the minimum connection count policy performs load balancing based on the following priority order: status and connection count.

Reviewer: ```suggestion
- `connection`: the minimum connection count priority policy performs load balancing based on the following priority order: status and connection count.
```

- When the TiDB cluster is deployed across data centers, to reduce network latency between TiProxy and TiDB servers, TiProxy prioritizes routing requests to TiDB servers in the same data center.

By default, this policy has a lower priority than health-based, memory-based, and CPU-based load balancing policies. You can increase its priority by setting [`policy`](/tiproxy/tiproxy-configuration.md#policy) to `location`. To maintain availability and performance, it is recommended to ensure that at least three TiDB servers are in the same location.

TiProxy determines the locations of itself and TiDB servers based on the `zone` label. You need to set the following configuration items:

- In the [`labels`](/tidb-configuration-file.md#labels) configuration item of the TiDB server, set `zone` to the current availability zone. For configuration details, see [Configure labels for TiDB](/schedule-replicas-by-topology-labels.md#optional-configure-labels-for-tidb).
- In the [`labels`](/tiproxy/tiproxy-configuration.md#labels) configuration item of TiProxy, set `zone` to the current availability zone.

> For clusters deployed using TiDB Operator, see [High availability of data
> ](https://docs.pingcap.com/tidb-in-kubernetes/stable/configure-a-tidb-cluster#high-availability-of-data).

Author: ```suggestion
For clusters deployed using TiDB Operator, see [High availability of data](https://docs.pingcap.com/tidb-in-kubernetes/stable/configure-a-tidb-cluster#high-availability-of-data).
```

## Connection count-based load balancing

TiProxy migrates connections from a TiDB server with more connections to a server with fewer connections. This policy is not configurable and has the lowest priority.

Typically, TiProxy identifies the load on TiDB servers based on CPU usage. This policy usually takes effect in the following scenarios:

- When the TiDB cluster is just starting up, the CPU usage of all TiDB servers is close to 0. In this case, this policy prevents imbalanced load during startup.
- When the [CPU-based load balancing](#cpu-based-load-balancing) is not enabled, this policy ensures load balancing.

## Configure load balancing policies

TiProxy lets you configure the combination and priority of load balancing policies through the [`policy`](/tiproxy/tiproxy-configuration.md#policy) configuration item.

- `resource`: the resource priority policy performs load balancing based on the following priority order: status, health, memory, CPU, location, and connection count.
- `location`: the location priority policy performs load balancing based on the following priority order: status, location, health, memory, CPU, and connection count.

> - `connection`: the minimum connection count priority policy performs load balancing based on the following priority order: status and connection count.

Reviewer: ```suggestion
- `connection`: the minimum connection count policy performs load balancing based on the following priority order: status and connection count.
```

Reviewer: see https://github.com/pingcap/docs-cn/pull/17485#discussion_r1639634511
